infping.go

```diff
@@ -4,156 +4,149 @@
 package main
 
 import (
-    "github.com/influxdata/influxdb/client"
-    "github.com/pelletier/go-toml"
-    "fmt"
-    "log"
-    "os"
-    "bufio"
-    "os/exec"
-    "net/url"
-    "strings"
-    "time"
-    "strconv"
+	"bufio"
+	"fmt"
+	client "github.com/influxdata/influxdb/client/v2"
+	"github.com/pelletier/go-toml"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
 )
 
 func herr(err error) {
-    if err != nil {
-        fmt.Println(err)
-        os.Exit(1)
-    }
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
 }
 
 func perr(err error) {
-    if err != nil {
-        fmt.Println(err)
-    }
+	if err != nil {
+		fmt.Println(err)
+	}
 }
 
 func slashSplitter(c rune) bool {
-    return c == '/'
+	return c == '/'
 }
 
-func readPoints(config *toml.TomlTree, con *client.Client) {
-    args := []string{"-B 1", "-D", "-r0", "-O 0", "-Q 10", "-p 1000", "-l"}
-    hosts := config.Get("hosts.hosts").([]interface{})
-    for _, v := range hosts {
-        host, _ := v.(string)
-        args = append(args, host)
-    }
-    log.Printf("Going to ping the following hosts: %q", hosts)
-    cmd := exec.Command("/usr/bin/fping", args...)
-    stdout, err := cmd.StdoutPipe()
-    herr(err)
-    stderr, err := cmd.StderrPipe()
-    herr(err)
-    cmd.Start()
-    perr(err)
-
-    buff := bufio.NewScanner(stderr)
-    for buff.Scan() {
-        text := buff.Text()
-        fields := strings.Fields(text)
-        // Ignore timestamp
-        if len(fields) > 1 {
-            host := fields[0]
-            data := fields[4]
-            dataSplitted := strings.FieldsFunc(data, slashSplitter)
-            // Remove ,
-            dataSplitted[2] = strings.TrimRight(dataSplitted[2], "%,")
-            sent, recv, lossp := dataSplitted[0], dataSplitted[1], dataSplitted[2]
-            min, max, avg := "", "", ""
-            // Ping times
-            if len(fields) > 5 {
-                times := fields[7]
-                td := strings.FieldsFunc(times, slashSplitter)
-                min, avg, max = td[0], td[1], td[2]
-            }
-            //log.Printf("Host:%s, loss: %s, min: %s, avg: %s, max: %s", host, lossp, min, avg, max)
-            writePoints(config, con, host, sent, recv, lossp, min, avg, max)
-        }
-    }
-    std := bufio.NewReader(stdout)
-    line, err := std.ReadString('\n')
-    perr(err)
-    log.Printf("stdout:%s", line)
+func readPoints(config *toml.TomlTree, con client.Client) {
+	args := []string{"-B 1", "-D", "-r0", "-O 0", "-Q 10", "-p 1000", "-l"}
+	hosts := config.Get("hosts.hosts").([]interface{})
+	for _, v := range hosts {
+		host, _ := v.(string)
+		args = append(args, host)
+	}
+	log.Printf("Going to ping the following hosts: %q", hosts)
+	cmd := exec.Command("/usr/bin/fping", args...)
+	stdout, err := cmd.StdoutPipe()
+	herr(err)
+	stderr, err := cmd.StderrPipe()
+	herr(err)
+	cmd.Start()
+	perr(err)
+
+	buff := bufio.NewScanner(stderr)
+	for buff.Scan() {
```

Comment:
Outside this loop is where you want your batch setup, ideally, then pass it into the writePoints method.